web/pages/flashcards/[setId].jsx

```diff
@@ -1,3 +1,7 @@
 import { PageSingleSet } from '@/lib/components/flashcards/PageSingleSet'
 
 export default PageSingleSet
+
+export function getServerSideProps({ query }) {
+  return { props: { setId: query.setId } }
+}
```

Comment:
Why do we have to pass `setId` from server side?

Reply:
using `const {  setId } = router.query` has certain delay, which conflicts with `useQuery`.
When `isLoading` is false, instead of return `flashcardSets` as array, it returns `flashcardSets` as undefined.
Passing props from serverside is more reliable as we have value of `setId` immediately